docs/DeveloperGuide.md

```diff
@@ -1,13 +1,304 @@
+- `/t` Reads the type of cashflow that the user wants to visualize (income/expense)
+- `/c` Reads the type of visualization tools the user wants (piechart/barchart)
+
+2) Calls the Cateorgizer to sort cashflow (Income/Expense) according to type
+
+3) Calls the Visualizer to display the chart to the user
+
+Categorizer's Role: 
+
+According to the cashflow type (Income/Expense) arugment passed in, the Categorizer sorts the 
+specified cashflow entry according to type using a Hashmap which is returned and used by the Visualizer
+
+Visualizer's Role: 
+
+According to the chart type (Pie/Bar/Radar) argument and the Hashmap obtained from the categorizer passed in, 
+the visualizer displays the specified visualization chart by calling the charting library Xchart.
+
+### Class Diagram
+
+![](images/vis/visualisationClass.png)
+
+### Sequence Diagram 
+
+Overall 
+
+![](images/vis/visualisationSequence.png)
+
+Categorizer
+
+![](images/vis/categorizerSequence.png)
+
+Visualizer
+
+![](images/vis/visualizerSequence.png)
+
+### Add Income/Expense Feature
+
+The add income/expense command has 2 compulsory arguments `/t` and `/a` and 1 optional argument `/r`.
+
+Example:
+```
+add income /a 100 /t salary /r 30
+```
+Below are the steps that shows the implementation of add income/expense.
+
+#### Step 1
+The AddCashflowCommand instance calls addIncome() or addExpense(), depending on what `category` is initialised as.
+
+addIncome() or addExpense() instantiates an Income or Expense object respectively.
+
+Example:
+```
+switch (category) {
+        case INCOME:
+            cashflowList.addIncome(amount, incomeType, recur);
+            break;
+        case EXPENSE:
+            cashflowList.addExpense(amount, expenseType, recur);
+            break;
+        default:
+            ui.showMessage("Unidentified entry.");
+            break;
+        }
+```
+#### Step 2
+The instantiated income/expense then updates the overall balance through addIncomeValue() or addExpenseValue().
+
+The income/expense object is also added to the list in Cashflowlist which contains all incomes/expenses.
+#### Step 3
+The added income/expense is then displayed to the user through the Ui.
+
+#### Add Income/Expense Class Diagram
+Given below is the class diagram showing the class structure of the add income/expense mechanism:
+![](images/cashflow/CashflowClassDiagram.png)
+#### Add Income/Expense Sequence Diagram
+Given below is the sequence diagram showing the add income/expense mechanism:
+![](images/cashflow/AddCashflowSequence.png)
+### Recurring Cashflow Feature
+Cashflow refers to an income or expense.
+
+This feature is called from the user through the `/r` argument in add income/expense command.
+
+If a cashflow is set to be recurring, the program would add another entry of the same cashflow to the Financial Planner after a set period of time.
+
+Below are the steps that shows the implementation of the recurring cashflow feature.
+
+#### Step 1
```

Comment:
I think it would be good to add a summary of each step
`#### Step 1: Store date of cashflow added`